### PR TITLE
Add some additional validation steps for Javascript

### PR DESCRIPTION
**Pull request by yeah568**
- Prevent user from using a clearly invalid URL
- Won't send invalid request if the user clicks cancel or an empty value in the prompt box.

### CODE REVIEW

File: extensions/gist.js [hunk 1]
```diff
@@ -37,7 +37,11 @@ function url_path_split(path) {
...
+}
+
+define(['jquery'], function ($) {
```
Comment (mreid-moz) [inline extensions/gist.js]: Is jquery actually being used?

File: extensions/gist.js [hunk 2]
```diff
@@ -57,8 +61,16 @@ define(function () {
     };
 
     var load_from_url = function() {
-        var url = prompt("Enter a Gist URL");
-        // TODO: check that it's a valid URL
+        var url = prompt("Enter a Gist or .ipynb file URL");
```
Comment (mreid-moz) [inline extensions/gist.js]: This is a bit unclear, maybe "Enter the URL to a Gist or a .ipynb"?

**Comment by mreid-moz** [comment]
@yeah568 Overall looks good, but see the 2 comments above

**Comment by yeah568** [comment]
@mreid-moz Whoops, forgot to push my commit. I've removed jQuery and changed the prompt text in the latest commit.

**Pull request merged**